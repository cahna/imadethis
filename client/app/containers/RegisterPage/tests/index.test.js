/**
 *
 * Tests for RegisterPage
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { Provider } from 'react-redux';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import { RegisterPage } from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';
import configureStore from '../../../configureStore';

describe('<RegisterPage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore();
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const onChangeUsername = jest.fn();
    const onChangePassword = jest.fn();
    const onChangeConfirmPassword = jest.fn();
    const onSubmitForm = jest.fn();
    const clearForm = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <RegisterPage
            username=""
            password=""
            confirmPassword=""
            usernameError={false}
            passwordError={false}
            confirmPasswordError={false}
            onChangeUsername={onChangeUsername}
            onChangePassword={onChangePassword}
            onChangeConfirmPassword={onChangeConfirmPassword}
            onSubmitForm={onSubmitForm}
            clearForm={clearForm}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it.skip('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <RegisterPage />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
