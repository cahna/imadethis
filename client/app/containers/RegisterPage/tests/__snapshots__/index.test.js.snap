// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterPage /> Should render and match the snapshot 1`] = `
<div>
  <section
    class="Section-ojdkzv-0 CenteredSection-sc-1e2by6n-0 fuuAJb"
  >
    <h2
      class="H2-egphza-0 fehwqM"
    >
      Register a new account:
    </h2>
  </section>
  <section
    class="Section-ojdkzv-0 CenteredSection-sc-1e2by6n-0 fuuAJb"
  >
    <form
      class="Form-kd3r69-0 fickPI"
    >
      <div>
        <label
          for="username"
        >
          Username
          <input
            class="Input-sc-19txxuq-0 bqjmLb"
            id="username"
            placeholder=""
            type="text"
            value=""
          />
        </label>
      </div>
      <div>
        <label
          for="password"
        >
          Password
          <input
            class="Input-sc-19txxuq-0 bqjmLb"
            id="password"
            placeholder=""
            type="password"
            value=""
          />
        </label>
      </div>
      <div>
        <label
          for="confirmPassword"
        >
          Confirm password
          <input
            class="Input-sc-19txxuq-0 bqjmLb"
            id="confirmPassword"
            placeholder=""
            type="password"
            value=""
          />
        </label>
      </div>
      <div
        class="Wrapper-sc-5dsnpv-0 bCngtA"
      >
        <input
          class="StyledInputButton-sc-114mby9-0 krjkFi"
          disabled=""
          type="submit"
        />
      </div>
    </form>
  </section>
</div>
`;
